src/haxelib/client/Main.hx

```diff
@@ -843,6 +846,39 @@ class Main {
 		Cli.print('Local repository deleted ($path)');
 	}
 
+	/** Outputs a lock file with all the installed libraries **/
+	function lock() {
+		if (RepoManager.getLocalPath(Sys.getCwd()) == null) {
+			Cli.printWarning("no local repository found, dumping lock from global");
+		}
+
+		final scope = getScope();
+		final libraryInfo = scope.getArrayOfLibraryInfo();
+
+		// sort projects alphabetically
+		libraryInfo.sort(function(a, b) return Reflect.compare(a.name.toLowerCase(), b.name.toLowerCase()));
+
+		for (library in libraryInfo) {
+			if (library.devPath != null) {
+				Cli.printError('Error: version of "${library.name}" is "dev". The lockfile will not work properly in other environments!');
+			}
+
+			final version = scope.getVersion(library.name);
```

Comment:
```diff
-                       final version = scope.getVersion(library.name);
+                       final version = try scope.getVersion(library.name) catch (e) {
+                               Cli.printError('Error: ${e.message}');
+                               continue;
+                       };
```

Dev libs that don't have "non dev" version installed will make `haxelib lock' abort early if you don't catch errors there.